### PR TITLE
Port the latest changes from Drupals htaccess blocks to Nginx rules

### LINKED ISSUE #817: Port Drupal blocks from #3308369 to Nginx configuration
See https://git.drupalcode.org/project/drupal/-/commit/2233484fbaabe30ecda8c15c49fed138b255be63 and https://www.drupal.org/project/drupal/issues/3308369 in which Drupal now blocks by default the following files.

```
yarn.lock
package.json
```

Originally raised in https://github.com/uselagoon/lagoon-images/issues/776#issuecomment-1700931371 by @thomasdik1989

### PR DESCRIPTION
**Pull request by seanhamlin**
Fixes #817.

### CODE REVIEW

File: images/nginx-drupal/drupal.conf [hunk 1]
```diff
@@ -36,7 +36,7 @@ server {
     ## Replicate the Apache <FilesMatch> directive of Drupal standard
     ## .htaccess. Disable access to any code files. Return a 404 to curtail
     ## information disclosure.
-    location ~* \.(engine|inc|install|make|module|profile|po|sh|.*sql|.*sql\.gz|theme|twig|tpl(\.php)?|xtmpl|yml)(~|\.sw[op]|\.bak|\.orig|\.save)?$|^\/(\.(?!well-known).*|Entries.*|Repository|Root|Tag|Template|web\.config)$|composer\.(json|lock)$|^\/#.*#$|\.php(~|\.sw[op]|\.bak|\.orig|\.save)$ {
+    location ~* \.(engine|inc|install|make|module|profile|po|sh|.*sql|.*sql\.gz|theme|twig|tpl(\.php)?|xtmpl|yml)(~|\.sw[op]|\.bak|\.orig|\.save)?$|^\/(\.(?!well-known).*|Entries.*|Repository|Root|Tag|Template|composer\.(json|lock)|web\.config)$|^\/#.*#$|\.php(~|\.sw[op]|\.bak|\.orig|\.save)|yarn.lock$|package.json$ {
```
Comment (sjerdo) [inline images/nginx-drupal/drupal.conf:39]: Can we include package-lock.json ?

```suggestion
    location ~* \.(engine|inc|install|make|module|profile|po|sh|.*sql|.*sql\.gz|theme|twig|tpl(\.php)?|xtmpl|yml)(~|\.sw[op]|\.bak|\.orig|\.save)?$|^\/(\.(?!well-known).*|Entries.*|Repository|Root|Tag|Template|composer\.(json|lock)|web\.config)$|^\/#.*#$|\.php(~|\.sw[op]|\.bak|\.orig|\.save)|yarn.lock$|package(-lock)?.json$ {
```

Comment (seanhamlin) [reply images/nginx-drupal/drupal.conf:39]: I don't see why not, it is not in stock Drupal core though https://github.com/drupal/drupal/blob/11.x/.htaccess - might be worth an upstream issue as well?

**Comment by seanhamlin** [comment]
Just adapted this from the latest https://github.com/drupal/drupal/blob/11.x/.htaccess rather than the original commit in the description.

File: images/nginx-drupal/drupal.conf [hunk 2]
```diff
@@ -36,7 +36,7 @@ server {
     ## Replicate the Apache <FilesMatch> directive of Drupal standard
     ## .htaccess. Disable access to any code files. Return a 404 to curtail
     ## information disclosure.
-    location ~* \.(engine|inc|install|make|module|profile|po|sh|.*sql|.*sql\.gz|theme|twig|tpl(\.php)?|xtmpl|yml)(~|\.sw[op]|\.bak|\.orig|\.save)?$|^\/(\.(?!well-known).*|Entries.*|Repository|Root|Tag|Template|web\.config)$|composer\.(json|lock)$|^\/#.*#$|\.php(~|\.sw[op]|\.bak|\.orig|\.save)$ {
+    location ~* \.(engine|inc|install|make|module|profile|po|sh|.*sql|.*sql\.gz|theme|twig|tpl(\.php)?|xtmpl|yml)(~|\.sw[op]|\.bak|\.orig|\.save)?$|^\/(\.(?!well-known).*|Entries.*|Repository|Root|Tag|Template|composer\.(json|lock)|web\.config|yarn\.lock|package(-lock)?\.json)$|^\/#.*#$|\.php(~|\.sw[op]|\.bak|\.orig|\.save)$ {
```
Comment (tobybellwood) [inline images/nginx-drupal/drupal.conf:39]: We modified the standard Drupal block slightly to block a full set of root dir items (up to web.config) - and then composer.json|lock anywhere in the path. I've modified this PR to block all of these remnant files wherever they occur

```suggestion
    location ~* \.(engine|inc|install|make|module|profile|po|sh|.*sql|.*sql\.gz|theme|twig|tpl(\.php)?|xtmpl|yml)(~|\.sw[op]|\.bak|\.orig|\.save)?$|^\/(\.(?!well-known).*|Entries.*|Repository|Root|Tag|Template)$|(composer\.(json|lock)|web\.config|yarn\.lock|package(-lock)?\.json)$|^\/#.*#$|\.php(~|\.sw[op]|\.bak|\.orig|\.save)$ {
```

Comment (sjerdo) [reply images/nginx-drupal/drupal.conf:39]: @seanhamlin Can you include this change?

Comment (tobybellwood) [reply images/nginx-drupal/drupal.conf:39]: I've tested locally, and I'm happy to include in the release this month if no-one objects!

**Pull request merged**